arcgis-ios-sdk-samples/Maps/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -24,31 +24,129 @@ class DisplayDeviceLocationWithNMEADataSourcesViewController: UIViewController {
         }
     }
+    /// The label to display accuracy info.
+    @IBOutlet var accuracyStatusLabel: UILabel!
     /// The label to display satellites info.
-    @IBOutlet var statusLabel: UILabel!
-    /// The button to start the demo.
-    @IBOutlet var startBarButtonItem: UIBarButtonItem!
+    @IBOutlet var satelliteStatusLabel: UILabel!
+    /// The button to choose a data source and start the demo.
+    @IBOutlet var sourceBarButtonItem: UIBarButtonItem!
     /// The button to reset pan mode to "recenter".
     @IBOutlet var recenterBarButtonItem: UIBarButtonItem!
     /// The button to reset the demo.
     @IBOutlet var resetBarButtonItem: UIBarButtonItem!
     
+    // MARK: Constants
+    
+    /// The protocols specified in the `Info.plist` that the app uses to
+    /// communicate with external accessory hardware.
+    let supportedProtocolStrings: Set<String> = Bundle.main
+        .object(forInfoDictionaryKey: "UISupportedExternalAccessoryProtocols")
+        .flatMap { $0 as? [String] }
+        .map(Set.init) ?? []
+    
     // MARK: Instance properties
     
     /// An NMEA location data source, to parse NMEA data.
-    let nmeaLocationDataSource = AGSNMEALocationDataSource(receiverSpatialReference: .wgs84())
+    var nmeaLocationDataSource: AGSNMEALocationDataSource?
     /// A mock data source to read NMEA sentences from a local file, and generate
     /// mock NMEA data every fixed amount of time.
     let mockNMEADataSource = SimulatedNMEADataSource(nmeaSourceFile: Bundle.main.url(forResource: "Redlands", withExtension: "nmea")!, speed: 1.5)
+    /// A formatter for the accuracy distance string.
+    let distanceFormatter: MeasurementFormatter = {
+        let formatter = MeasurementFormatter()
+        formatter.unitOptions = .naturalScale
+        formatter.numberFormatter.minimumFractionDigits = 1
+        formatter.numberFormatter.maximumFractionDigits = 1
+        return formatter
+    }()
     
     // MARK: Actions
     
-    @IBAction func start() {
+    /// Get the first connected and supported Bluetooth accessory with its
+    /// protocol string.
+    /// - Returns: A tuple of the accessory and its protocol,
+    ///            or nil if no supported accessory exists.
+    func firstSupportedAccessoryWithProtocol() -> (EAAccessory, String)? {
+        for accessory in EAAccessoryManager.shared().connectedAccessories {
+            // The protocol string to establish the EASession.
+            guard let protocolString = accessory.protocolStrings.first(where: { supportedProtocolStrings.contains($0) }) else {
+                // Skip any device which protocol is not included in the plist.
+                // This typically shouldn't happen, unless the device requires
+                // additional configuration.
+                continue
+            }
+            // Only return the first connected and supported accessory.
+            return (accessory, protocolString)
+        }
+        return nil
+    }
```

Comment:
Chatted with Eric Ito:

> Yes it is possible. We only use the one that emits NMEA but some devices support multiple and you could, in theory, read from them.

It goes back to https://github.com/Esri/arcgis-runtime-samples-ios/pull/1058#discussion_r616010708 and I think the future-proof approach would be to hardcode them in the sample, so that we only have 1 NMEA protocol for each device.

Was your comment cut at the end? I'm curious for what after "if not..." 😳

Reply:
I started to say say something, but then changed my mind and apparently forgot to retract what I started to say! I was going to say that if not, we may need a more deterministic method of picking protocols as I don't know whether the order of `protocolStrings` is deterministic.